core/src/services/dropbox/backend.rs

```diff
@@ -90,4 +91,38 @@ impl Accessor for DropboxBackend {
             _ => Err(parse_error(resp).await?),
         }
     }
+
+    async fn stat(&self, path: &str, _: OpStat) -> Result<RpStat> {
+        if path == "/" {
+            return Ok(RpStat::new(Metadata::new(EntryMode::DIR)));
+        }
+        let resp = self.core.dropbox_get_stat(path).await?;
+        let status = resp.status();
+        if status.is_success() {
+            let bytes = resp.into_body().bytes().await?;
+            let decoded_response = serde_json::from_slice::<DropboxMetadataResponse>(&bytes)
+                .map_err(new_json_deserialize_error)?;
+            let entry_mode: EntryMode = match decoded_response.tag.as_str() {
+                "file" => EntryMode::FILE,
+                "folder" => EntryMode::DIR,
+                _ => EntryMode::Unknown,
+            };
+            let mut metadata = Metadata::new(entry_mode);
+            let last_modified = decoded_response.client_modified;
+            let date_utc_last_modified = parse_datetime_from_rfc3339(&last_modified)?;
+            metadata.set_last_modified(date_utc_last_modified);
+            if decoded_response.size.is_some() {
+                let size = decoded_response.size.unwrap();
+                metadata.set_content_length(size);
+            }
+            Ok(RpStat::new(metadata))
+        } else {
+            match status {
```

Comment:
I don't like code like `if status { xxx } else { match status { yyy } }`. How about writing them in flat way?